### PR TITLE
Use `react-form-hook` v7 for flow form

### PR DESCRIPTION
**Pull request by jonkoops**
Refactors the flow form to use v7 of `react-form-hook`, and cleans up the code a little.

### CODE REVIEW

File: apps/admin-ui/src/authentication/DuplicateFlowModal.tsx [hunk 1]
```diff
@@ -31,27 +32,25 @@ export const DuplicateFlowModal = ({
...
 
   useEffect(() => {
-    setValue("description", description);
     setValue("alias", t("copyOf", { name }));
-  }, [name, description, setValue]);
+    setValue("description", description);
+  }, [name, description]);
 
-  const save = async () => {
+  const onSubmit = async () => {
```
Comment (edewit) [inline apps/admin-ui/src/authentication/DuplicateFlowModal.tsx:47]: forgot to point this out in the previous PR,  but why change this to `onSubmit` that is not what is happening here, save is a better name. As `handleSubmit` from react hook form is handling the on submit

File: apps/admin-ui/src/authentication/DuplicateFlowModal.tsx [hunk 2]
```diff
@@ -82,33 +81,34 @@ export const DuplicateFlowModal = ({
...
       onClose={toggleDialog}
       variant={ModalVariant.small}
       actions={[
         <Button
-          id="modal-confirm"
           key="confirm"
-          onClick={save}
           data-testid="confirm"
+          type="submit"
+          form="duplicate-flow-form"
```
Comment (edewit) [inline apps/admin-ui/src/authentication/DuplicateFlowModal.tsx:91]: :+1:

File: apps/admin-ui/src/authentication/DuplicateFlowModal.tsx (hunk 1 above)
Comment (jonkoops) [reply apps/admin-ui/src/authentication/DuplicateFlowModal.tsx:47]: The reason why I chose this approach is that it's inline with the examples from the React Form Hook documentation, which I try to follow as closely as possible.

We're also using different names for these callbacks in an inconsistent manner throughout the code-base, so this is an effort to make those names consistent throughout.

File: apps/admin-ui/src/authentication/DuplicateFlowModal.tsx (hunk 2 above)
Comment (jonkoops) [reply apps/admin-ui/src/authentication/DuplicateFlowModal.tsx:91]: Yeah, this is much better for accessibility. Found this gem in some of Jenny's code, now I am trying to apply it everywhere.

**Review by edewit — APPROVED** [review]

**Review by agagancarczyk — COMMENTED** [review]
LGTM

**Pull request merged**